### PR TITLE
NEXUS-6246: Checksum search fails after repair index task is run

### PR DESCRIPTION
**Pull request by cstamas**
Reason is that MI context producer will find no hashes in proxies
as NX is changed to store those in attributes. Adding an override
content producer that hints hashes getting them from NX in case
of M2 proxy repositories.

Issue
https://issues.sonatype.org/browse/NEXUS-6246

CI
http://bamboo.s/browse/NX-OSSF493

### CODE REVIEW

**Comment by peterlynch** [comment]
any idea how much this slows down reindex? I think I recall cases where it takes over 24hrs in some instances, so if this slows it down quite a bit I am worried by this change. People turn reindex on for fun...

**Comment by mcculls** [comment]
+1 looks reasonable

File: plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+public class NexusArtifactContextProducer
+    extends DefaultArtifactContextProducer
+{
+  private final Logger log;
+
+  private final RepositoryRegistry repositoryRegistry;
+
+  @Inject
+  public NexusArtifactContextProducer(final RepositoryRegistry repositoryRegistry) {
+    this.log = LoggerFactory.getLogger(getClass());
```
Comment (kellyrob99) [inline plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java]: minor, but this doesn't need to be in the constructor

File: plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  @Override
+  public ArtifactContext getArtifactContext(final IndexingContext context, final File file) {
+    final ArtifactContext ac = super.getArtifactContext(context, file);
+    if (ac != null && ac.getArtifactInfo().sha1 == null) {
+      // special case for proxies
+      String itemPath = file.getPath();
+      try {
+        // NOTE: MI supports Maven2 repositories only, so if this line does not throw,
```
Comment (kellyrob99) [inline plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java]: What is "MI please?

Comment (mcculls) [inline plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java]: Maven Indexer

File: plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        // we know we deal with a Maven2 proxy repository (no need to check is it Maven1)
+        final MavenProxyRepository proxyRepository = repositoryRegistry.getRepositoryWithFacet(context.getRepositoryId(),
+            MavenProxyRepository.class);
+        final File baseDir = getRepositoryLocalStorageFile(proxyRepository, RepositoryItemUid.PATH_ROOT);
+        itemPath = itemPath.substring(baseDir.getPath().length());
```
Comment (kellyrob99) [inline plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java]: baseDir here could be null according to getRepositoryLocalStorageFile() method

File: plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java (hunk 2 above)
Comment (kellyrob99) [inline plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/mindexer/NexusArtifactContextProducer.java]: tyvm @mcculls :)

**Comment by kellyrob99** [comment]
+1

**Comment by cstamas** [comment]
Manually merged as b72936fe419e8fa2c95932f4e1031bdff06da255

**Pull request closed without merging**